From 3d7bf164da30b1ad9cb4e912d4e9871042e96796 Mon Sep 17 00:00:00 2001
From: Aryan Khare <aryakhar@qti.qualcomm.com>
Date: Thu, 12 Feb 2026 17:59:10 +0530
Subject: [PATCH] Fix for BTM tests [test_80211v] on Airties TH

Enables bss_transition, disassoc_imminent and other flags related to BTM
with handling the UCI persistency for each field.

Signed-off-by: Aryan Khare <aryakhar@qti.qualcomm.com>
---
 agent/config/beerocks_agent.conf.in                 | 6 +++---
 common/beerocks/bwl/include/bwl/ap_wlan_hal.h       | 2 +-
 common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp | 6 ++++++
 framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp     | 8 +++++++-
 4 files changed, 17 insertions(+), 5 deletions(-)

diff --git a/agent/config/beerocks_agent.conf.in b/agent/config/beerocks_agent.conf.in
index 7a2ce96c7..05ab6feef 100644
--- a/agent/config/beerocks_agent.conf.in
+++ b/agent/config/beerocks_agent.conf.in
@@ -16,7 +16,7 @@ enable_arp_monitor=0
 bridge_iface=@BEEROCKS_BRIDGE_IFACE@
 enable_system_hang_test=0 # 0 - disabled
 monitor_polling_rate_msec=@BEEROCKS_MONITOR_POLLING_RATE_MSEC@
-enable_auto_chansel_handling=0 # 0 - disabled
+enable_auto_chansel_handling=1 # 0 - disabled
 
 # This variable is a workaround that turn on the profile_x_disallow flag when configuring the
 # traffic separation, if the Controller sent unsupported configuration on the profile disallow
@@ -58,11 +58,11 @@ dwell_time=20
 
 # Indicates whether unicast traffic statistics are collected per station
 # 1 = enable
-clients_unicast_measurements=0
+clients_unicast_measurements=1
 
 # send_btm_to_non_11v_sta = 0 -> Do not send BTM requests to STAs that do not support 802.11v
 # send_btm_to_non_11v_sta = 1 -> Allow sending BTM requests to STAs even if they do not support 802.11v
-send_btm_to_non_11v_sta=0
+send_btm_to_non_11v_sta=1
 
 # Traffic Separation Configuration
 # TrafficSeparation_bridge_mapping_enabled = 0 -> Bridge mapping disabled (interfaces not moved to respective bridges)
diff --git a/common/beerocks/bwl/include/bwl/ap_wlan_hal.h b/common/beerocks/bwl/include/bwl/ap_wlan_hal.h
index c42973125..d150122d3 100644
--- a/common/beerocks/bwl/include/bwl/ap_wlan_hal.h
+++ b/common/beerocks/bwl/include/bwl/ap_wlan_hal.h
@@ -94,7 +94,7 @@ public:
         int reason              = 0;
         bool pref_list_included = 0;
         bool abridged           = 0;
-        bool disassoc_imminent  = 0;
+        bool disassoc_imminent  = 1;
     };
 
     // Public methods
diff --git a/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp b/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
index 4f9ba54d9..d4c7b9640 100644
--- a/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
+++ b/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
@@ -1717,6 +1717,9 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
                     }
                 }
                 conf.set_create_vap_value(vap_id, "multi_ap", multi_ap);
+                conf.set_create_vap_value(vap_id, "bss_transition", 1);
+                conf.set_create_vap_value(vap_id, "rrm_neighbor_report", 1);
+                conf.set_create_vap_value(vap_id, "rrm_beacon_report", 1);
                 //required to update uci wireless
                 bss_it.multi_ap = multi_ap;
                 // oddly enough, multi_ap_backhaul_wpa_passphrase has to be
@@ -3235,6 +3238,9 @@ bool ap_wlan_hal_nl80211::create_new_hostapd_conf_file(std::string &ifname, son:
     conf.set_create_vap_value(vap_id,"notify_mgmt_frames", "1");
     conf.set_create_vap_value(vap_id,"ctrl_interface",
                                std::string(bwl::nl80211::base_ctrl_path) + "hostapd");
+    conf.set_create_vap_value(vap_id, "bss_transition", 1);
+    conf.set_create_vap_value(vap_id, "rrm_neighbor_report", 1);
+    conf.set_create_vap_value(vap_id, "rrm_beacon_report", 1);
 
     if (!assign_auth_encr_parameters(conf,vap_id, bss_conf)) {
         LOG(ERROR) << "Failed to set the authentication/encryption parameters!";
diff --git a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
index 6f827fe5a..e1ba54b3f 100644
--- a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
+++ b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
@@ -507,7 +507,10 @@ bool bpl_cfg_set_wifi_credentials(const std::string &iface,
 		    options["multi_ap_backhaul_ssid"] = configuration.backhaul_wps_ssid;
     		    options["multi_ap_backhaul_key"]  = configuration.backhaul_wps_passphrase;
 	    }
-	    
+
+        options["bss_transition"] = "1";
+        options["ieee80211k"] = "1";
+
 	    std::string device_section;
 	    
 	    if (!radio_id.empty() &&
@@ -611,6 +614,9 @@ bool bpl_cfg_set_wifi_credentials(const std::string &iface,
                 options["multi_ap_backhaul_key"] = config.backhaul_wps_passphrase;
             }
 
+            options["bss_transition"] = "1";
+            options["ieee80211k"] = "1";
+
             return options;
         };
 
-- 
2.34.1

